pkg/spec/config_linux.go

```diff
@@ -46,18 +46,20 @@ func devicesFromPath(g *generate.Generator, devicePath string) error {
 		return errors.Wrapf(err, "cannot stat %s", devicePath)
 	}
 	if st.IsDir() {
-		if len(devs) > 2 {
-			return errors.Wrapf(unix.EINVAL, "not allowed to specify destination with a directory %s", devicePath)
-		}
 		found := false
 		// mount the internal devices recursively
 		if err := filepath.Walk(resolvedDevicePath, func(dpath string, f os.FileInfo, e error) error {
 
 			if f.Mode()&os.ModeDevice == os.ModeDevice {
 				found = true
 				device := dpath
-
-				if len(devs) > 1 {
+				if len(devs) > 1 && devs[1][0] == '/' {
```

Comment:
Can we add a comment to that? I wasn't familiar with those specific parts and it took me a while to understand it. Although the compiler might be smart enough to move `len(devs) > 1 && devs[1][0] == '/'` outside the walk func, can we move the checks outside the walk and assign them to some booleans that describe it a bit more?

Something as follows (just an example) might be easier to read:
```golang
destSpecified := len(devs) > 1 && devs[1][0] == '/'
modeSpecified := len(devs) > 1 && devs[1][0] != '/'
destAndModeSpecified := len(devs) > 2
destOrModeSpecified := len(devs) > 1
// mount the internal devices recursively
if err := filepath.Walk(resolvedDevicePath, fun [...]
```